src/components/App.tsx

```diff
@@ -18,13 +25,14 @@ const AppContent = styled.div`
   }
 `;
 
-class App extends Component {
-  constructor(props) {
+class App extends Component<Props, State> {
+  constructor(props: Props) {
     super(props);
     this.state = {
       data: {
         pages: {
           pageData: {
+            // @ts-ignore: This doesn't match the JSONSchema!!!
```

Comment:
@sarah-johnson: This doesn't match the data schema that you gave me, so TypeScript complains. Maybe this needs updating?

Reply:
To be sure. I'm sure there's a more elegant way to handle this case. I have included this block in order for the application to not crash prior to getting the initial API response. There must be a more elegant way - Just writing "Loading Karst" to the screen until the API responds would be great, for example.

Reply:
We removed this and replaced it with a "Loading Karst" message